Makefile

```diff
@@ -99,10 +99,10 @@ $(GEN_DIR):
 $(GEN_SRCS): dbus-vmstate1.xml | $(GEN_DIR)
 	gdbus-codegen --interface-prefix org.qemu. --output-directory $(GEN_DIR) --generate-c-code dbus_vmstate1 --c-generate-object-manager $<
 
-$(BUILD_DIR_GEN)/%.o: $(GEN_DIR)/%.c | $(BUILD_DIR_GEN)
+$(BUILD_DIR_GEN)/%.o: $(GEN_DIR)/%.c $(GEN_SRCS)| $(BUILD_DIR_GEN)
```

Comment:
I don't think this is right; GEN_SRCS still only refers to the .c file (if you touch the .h I bet it's not re-generated), and also, you now shouldn't need $(GEN_DIR)/%.c here either?

Reply:
the thing is gdbus-codegen will generate both dbus-vmstate.c and .c in gen/ folder.

Reply:
And we require $(GEN_DIR)/%.c to get the correct c file in $< while in the build.

Reply:
I know that gdbus-codegen does that, but you're not reflecting that in your dependencies. Imagine if it re-ran and *only* updated the header, but that mattered for some reason.

$< will still pick the first in $(GEN_SRCS) so I don't think you need both?

Reply:
ok. got it.
Added dbus_vmstate1.h to the dependencies list.